overlord/snapstate/handlers.go

```diff
@@ -1007,9 +1007,11 @@ func maybeRestart(t *state.Task, info *snap.Info) {
 		logger.Noticef("cannot get model assertion: %v", model)
 		return
 	}
-	if model.Base() != "" && info.InstanceName() == "snapd" {
-		t.Logf("Requested daemon restart (snapd snap).")
-		st.RequestRestart(state.RestartDaemon)
+	if info.InstanceName() == "snapd" {
```

Comment:
this means we plan for people to try the transition also on core? not only classic for now?  also if we do this change do we need to get the Model some lines above?

Reply:
Indeed, we should limit this to classic for now. I will update the code.